src/Appwrite/Platform/Modules/Proxy/Http/Rules/Create.php

```diff
@@ -0,0 +1,195 @@
+                    )
+                ]
+            ))
+            ->label('abuse-limit', 10)
+            ->label('abuse-key', 'userId:{userId}, url:{url}')
+            ->label('abuse-time', 60)
+            ->param('domain', null, new ValidatorDomain(), 'Domain name.')
+            ->param('resourceType', null, new WhiteList(['api', 'function', 'site']), 'Action definition for the rule. Possible values are "api", "function" and "site"')
+            ->param('resourceId', '', new UID(), 'ID of resource for the action type. If resourceType is "api", leave empty. If resourceType is "function", provide ID of the function.', true)
+            ->inject('response')
+            ->inject('project')
+            ->inject('queueForCertificates')
+            ->inject('queueForEvents')
+            ->inject('dbForPlatform')
+            ->inject('dbForProject')
+            ->callback([$this, 'action']);
+    }
+
+    public function action(string $domain, string $resourceType, string $resourceId, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform, Database $dbForProject)
+    {
+        $mainDomain = System::getEnv('_APP_DOMAIN', '');
+        if ($domain === $mainDomain) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your main domain to specific resource. Please use subdomain or a different domain.');
+        }
+
+        $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
+
+        if ($domain === 'localhost' || $domain === APP_HOSTNAME_INTERNAL) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'This domain name is not allowed. Please pick another one.');
+        }
+
+        // TODO: @christyjacob remove once we migrate the rules in 1.7.x
+        if (System::getEnv('_APP_RULES_FORMAT') === 'md5') {
+            $document = $dbForPlatform->getDocument('rules', md5($domain));
+        } else {
+            $document = $dbForPlatform->findOne('rules', [
+                Query::equal('domain', [$domain]),
+            ]);
+        }
+
+        if (!$document->isEmpty()) {
+            if ($document->getAttribute('projectId') === $project->getId()) {
+                $resourceType = $document->getAttribute('resourceType');
+                $resourceId = $document->getAttribute('resourceId');
+                $message = "Domain already assigned to another resource.";
+                if (!empty($resourceId)) {
+                    $message .= " with ID '{$resourceId}'";
+                }
+
+                $message .= '.';
+            } else {
+                $message = 'Domain already assigned to different project.';
+            }
+
+            throw new Exception(Exception::RULE_ALREADY_EXISTS, $message);
+        }
+
+        $resourceInternalId = '';
+
+        switch ($resourceType) {
```

Comment:
This switch/case duplicates too much code. Only thing that is really changing is collection name, which can be a varaible. We can avoid this switch/case